src/Servers/Kestrel/Core/src/Internal/Http2/Http2OutputProducer.cs

```diff
@@ -405,10 +410,6 @@ private async ValueTask<FlushResult> ProcessDataWrites()
                     _pipeReader.AdvanceTo(readResult.Buffer.End);
                 } while (!readResult.IsCompleted);
             }
-            catch (OperationCanceledException)
```

Comment:
Nothing else in this loop throws this exception right? I see we pass a default token to `_frameWriter.FlushAsync`. If we're not sure, it might be less risky to leave this catch or we'll end up logging a critical error.

@halter73?

Reply:
I took a look and I don't believe anywhere can throw this exception. WriteDataAsync doesn't take a cts and FlushAsync is default. However, if you prefer to be defensive with this change, I can readd the OperationCanceledException

Reply:
Might be better to be more defensive in 3.0 but remove it in master.

Reply:
But if we're very sure then i'm OK with it.

Reply:
Nothing here can throw an OperationCanceledException after removing that call to complete. I feel good removing this.